src/main/java/com/kakao/golajuma/vote/domain/service/GetVoteListService.java

```diff
@@ -60,53 +57,93 @@ public GetVoteListResponse.MainAndFinishPage getVoteList(
+		return completeByCategoryOrderBySort(getCategory(category), sort);
 	}
 
-	private Category checkCategory(String category) {
-		return Category.findCategory(category);
+	private Slice<VoteEntity> completeOrderBySort(String sort) {
+		// 어디서부터 몇개씩 가져올건지
+		Pageable pageable = PageRequest.of(page, size);
+		LocalDateTime now = LocalDateTime.now();
+
+		if (isCurrent(sort)) {
+			return voteRepository.findAllFinishVotesOrderByCreatedDate(now, pageable);
+		}
+		return voteRepository.findAllFinishVotesOrderByVoteTotalCount(now, pageable);
 	}
 
-	private Slice<VoteEntity> findContinueVotes(String sort, Category category) {
+	private Slice<VoteEntity> completeByCategoryOrderBySort(Category category, String sort) {
 		// 어디서부터 몇개씩 가져올건지
 		Pageable pageable = PageRequest.of(page, size);
 
 		LocalDateTime now = LocalDateTime.now();
 
-		if (sort.equals("current")) {
-			return voteRepository.findAllContinueVotesByCategoryOrderByCreatedDate(
-					now, category, pageable);
-		}
-		if (sort.equals("popular")) {
-			return voteRepository.findAllContinueVotesByCategoryOrderByVoteTotalCount(
-					now, category, pageable);
+		if (isCurrent(sort)) {
+			return voteRepository.findAllFinishVotesByCategoryOrderByCreatedDate(now, category, pageable);
 		}
+		return voteRepository.findAllFinishVotesByCategoryOrderByVoteTotalCount(
+				now, category, pageable);
+	}
 
-		throw new RequestParamException("잘못된 요청입니다.(sort)");
+	private Slice<VoteEntity> findContinueVotes(String category, String sort) {
+		if (getCategory(category) == Category.TOTAL) {
+			return continueOrderBySort(sort);
+		}
+		return continueByCategoryOrderBySort(getCategory(category), sort);
 	}
 
-	private Slice<VoteEntity> findFinishVotes(String sort, Category category) {
+	private Slice<VoteEntity> continueOrderBySort(String sort) {
 		// 어디서부터 몇개씩 가져올건지
 		Pageable pageable = PageRequest.of(page, size);
 
 		LocalDateTime now = LocalDateTime.now();
 
-		if (sort.equals("current")) {
-			return voteRepository.findAllFinishVotesByCategoryOrderByCreatedDate(now, category, pageable);
+		if (isCurrent(sort)) {
+			return voteRepository.findAllContinueVotesOrderByCreatedDate(now, pageable);
 		}
-		if (sort.equals("popular")) {
-			return voteRepository.findAllFinishVotesByCategoryOrderByVoteTotalCount(
+		return voteRepository.findAllContinueVotesOrderByVoteTotalCount(now, pageable);
+	}
+
+	private Slice<VoteEntity> continueByCategoryOrderBySort(Category category, String sort) {
+		// 어디서부터 몇개씩 가져올건지
+		Pageable pageable = PageRequest.of(page, size);
+
+		LocalDateTime now = LocalDateTime.now();
+
+		if (isCurrent(sort)) {
+			return voteRepository.findAllContinueVotesByCategoryOrderByCreatedDate(
 					now, category, pageable);
 		}
+		return voteRepository.findAllContinueVotesByCategoryOrderByVoteTotalCount(
+				now, category, pageable);
+	}
+
+	private boolean isCurrent(String sort) {
+		return Sort.findSort(sort) == Sort.CURRENT;
+	}
+
+	private boolean isContinueRequest(String active) {
```

Comment:
Actvie 가 isConitnue() 메시지를 받아서
isContinue에서 active를 인자로 받아
Active에서 본인의 상태를 본인이 검증하게 해도 좋을 듯 합니다!

Reply:
```suggestion
Active{
// 생략

 public  boolean isContinueRequest(String active) {
     Category category = Category.findCategory(category);
     if(category.eqauls(Active.COMPLETE)){
         return true;
      }
      return false;
 }
```

[데미테르 법칙](https://mangkyu.tistory.com/147)
상태를 요청하지 말고 요구하라